x-pack/plugins/security_solution/server/usage/queries/utils/fetch_hits_with_pit.ts

```diff
@@ -0,0 +1,79 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import type {
+  OpenPointInTimeResponse,
+  SearchHit,
+  SortResults,
+} from '@elastic/elasticsearch/lib/api/typesWithBodyKey';
+import type { SearchRequest } from '@elastic/elasticsearch/lib/api/types';
+import type { ElasticsearchClient, Logger } from 'kibana/server';
+
+export interface FetchWithPitOptions {
+  esClient: ElasticsearchClient;
+  index: string;
+  maxSize: number;
+  maxPerPage: number;
+  searchRequest: SearchRequest;
+  logger: Logger;
+}
+
+export const fetchHitsWithPit = async <T>({
+  esClient,
+  index,
+  searchRequest,
+  maxSize,
+  maxPerPage,
+  logger,
+}: FetchWithPitOptions): Promise<Array<SearchHit<T>>> => {
+  // default is from looking at Kibana saved objects and online documentation
+  const keepAlive = '5m';
+
+  // create and assign an initial point in time
+  let pitId: OpenPointInTimeResponse['id'] = (
+    await esClient.openPointInTime({
+      index,
+      keep_alive: '5m',
+    })
+  ).id;
+
+  let searchAfter: SortResults | undefined;
+  let hits: Array<SearchHit<T>> = [];
+  let fetchMore = true;
+  while (fetchMore) {
+    const ruleSearchOptions: SearchRequest = {
+      ...searchRequest,
+      track_total_hits: false,
+      search_after: searchAfter,
+      sort: [{ _shard_doc: 'desc' }] as unknown as string[], // Remove this "unknown" once it is typed correctly https://github.com/elastic/elasticsearch-js/issues/1589
```

Comment:
Some IDEs have todo scanning like intelliJ. If you but `TODO: Remove this "unknown"...` It will show up in people's consoles. It's always a neat trick I like to employ

Reply:
I can add a `TODO` for this. 👍